spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +114,53 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+      table: DeltaTableV2,
+      target: String,
+      replace: Boolean,
+      properties: Map[String, String],
+      versionAsOf: Option[Long] = None,
+      timestampAsOf: Option[String] = None): DeltaTable = withActiveSession(sparkSession) {
+    val (targetRelation, targetPath) = if (new Path(target).isAbsolute()) {
+      (UnresolvedRelation(TableIdentifier(target, Some("delta"))), Some(target))
+    } else {
+      val tableIdentifier = sparkSession
+        .sessionState
+        .sqlParser
+        .parseTableIdentifier(target)
+      (UnresolvedRelation(tableIdentifier), None)
+    }
+
+    val sourceIdentifier = table.getTableIdentifierIfExists.map(id =>
+      Identifier.of(id.database.toArray, id.table))
+    val sourceRelation = DataSourceV2Relation.create(table, None, sourceIdentifier)
+
+    val maybeTimeTravelSource = if (timestampAsOf.isDefined || versionAsOf.isDefined) {
+      TimeTravel(
+        sourceRelation,
+        timestampAsOf.map(Literal(_)),
+        versionAsOf,
+        None
+      )
+    } else {
+      sourceRelation
+    }
+
+    val clone = CloneTableStatement(maybeTimeTravelSource, targetRelation, false, replace, true,
+      properties.toMap, targetPath)
+
+    val qe = sparkSession.sessionState.executePlan(clone)
+
+    // call `QueryExecution.toRDD` to trigger the execution of commands.
+    SQLExecution.withNewExecutionId(qe, Some("clone delta table"))(qe.toRdd)
+
+    if (targetPath.isDefined) {
```

Comment:
`targetPath.isDefined` is the same as `new Path(target).isAbsolute()`

Let's put a single variable in the function to avoid repitition `val assumePath = new Path(target).isAbsolute()`

Reply:
What do you mean? The whole reason I'm checking `targetPath.isDefined` is to avoid repetition?

Reply:
@Kimahriman I mean let's put one variable down

`val assumePath = new Path(target).isAbsolute()`

Then use it twice.

Reply:
target path isn't a thing anymore so made a variable